lib/client.go

```diff
@@ -27,28 +30,127 @@ type Client struct {
 	// ServerIp is the public IPv4 address of the server.
 	ServerIp netip.Addr
 
-	// Password is authenticates the client connection.
+	// Password authenticates the client connection.
 	Password string
 
 	// WgClient is a shared client for interacting with the WireGuard kernel module.
 	WgClient *wgctrl.Client
 
 	// Http is used to make connect requests to the server.
 	Http *http.Client
+
+	// serverPublicKey is the public key for the wireguard connection
+	serverPublicKey wgtypes.Key
+
+	// wgLink is the wireguard link
+	wgLink *linkWireguard
+
+	// wgCidr is the current wgCidr used with the wireguard link
+	wgCidr netip.Prefix
+}
+
+// todo: not sure if this is the right name (because now DeleteInterface dosen't seem so paired with it)
+// Connect attempts to set up the connection with the peer and
+// create a network interface for it.
+func (c *Client) Connect() error {
+	connectionResponse, err := c.sendConnectionRequest()
+	if err != nil {
+		return err
+	}
+
+	err = c.createInterface(connectionResponse)
+	if err != nil {
+		return err
+	}
```

Comment:
We discussed this before but you could split out CreateInterface for state handling, and then combine Connect() and Reconnect() into a single function.

Reply:
I did end up sort of splitting out createInterface -- the old `CreateInterface()` used to handle setting up the connection request, creating the interface, _and_ setting up wireguard. I renamed that to what's now `Connect()`, and extracted those three steps into separate helpers, including the current version of `createInterface()` which is much narrower.

However, I'm not sure what a good way to combine `Connect()` and `Reconnect()` any further would be, because the functionality that differs is in the middle (they both call the same helpers to send a connection request and configure wireguard, but `Connect()` also needs to create the network interface in between those two calls). `Connect()` also cleans up the interface it creates on failure, whereas `Reconnect()` won't touch the existing interface on failure.

Reply:
You can move the interface creation out of Connect() and into a CreateInterface() function, this has the advantage of creating the interface earlier after process startup and simplifying error handling.

Reply:
Ah I see what you mean, this should be fixed now.